### DISCUSSION TITLE
How to make rectangle button do "toggle full screen" under KDE Plasma 6?

### DISCUSSION THREAD

**Discussion by HubKing** [tomasklaen/uosc#878] [Q&A] [answered]
![image](https://github.com/tomasklaen/uosc/assets/10900489/e5e1868b-4e3c-4110-9723-b7cc84431bf3)

The button toggled full-screen under Gnome, but under KDE 6, it acts exactly like the maximise button in the window title bar.

**Comment by tomasklaen** [comment, marked as answer]
See #811 and #203.

In summary, we try to make this button maximize the window, and fall back to fullscreen where it is buggy.

As an alternative, controls bar has a shorthand for `fullscreen` control.

**Reply by HubKing** [reply]
Wish there was a way to explicitly choose the behaviour of the square button (either maximise or full-screen) or a way to put the full-screen button where the 3 window control buttons are, like other players.

![image](https://github.com/tomasklaen/uosc/assets/10900489/7b2d15cd-20f7-47d0-a88f-b5d557a3da19)
Gom Player

![image](https://github.com/tomasklaen/uosc/assets/10900489/6d970cf6-4a2d-44a1-a558-37923cd1dbc0)
Pot Player

**Reply by hooke007** [reply]
And then adding more buttons on the top bar?

**Reply by HubKing** [reply]
I didn't think it is possible to customise that part. There was top_bar_controls, but when I changed it from "yes" to "menu", the UI did not change. How to add a fullscreen button there?

```
top_bar_controls=menu
# Can be: `no` (hide), `yes` (inherit title from mpv.conf), or a custom template string
```

**Reply by HubKing** [reply]
Seems like I misread the comment. The "custom template" is just a string formatter for the title, not custom controls like the thing above the timeline. In the end, I just edited TopBar.lua and changed the 73th line to `			command = 'cycle fullscreen',`.
